tests/test_api.py

```diff
@@ -310,6 +311,54 @@ def test_upload_download(self):
         # cleanup
         os.remove(file_path)
 
+    @patch('shotgun_api3.Shotgun._send_form')
+    def test_upload_to_sg(self, mock_send_form):
+        """
+        Upload an attachment tests for _upload_to_sg()
+        """
+        mock_send_form.method.assert_called_once()
+        mock_send_form.return_value = "1\n:123\nasd"
+        if 'localhost' in self.server_url:
+            print("upload / down tests skipped for localhost")
+            return
```

Comment:
Another option could be to use pytest's [skipif](https://docs.pytest.org/en/latest/how-to/skipping.html#id1) marker.

Not a blocker, just to let you know you have this option.

Reply:
Agreed. Either a decorator or just call [skipIf](https://docs.python.org/3/library/unittest.html#unittest.skipIf):

```suggestion
        self.skipIf(
            'localhost' in self.server_url,
            "upload / down tests skipped for localhost",
        )
```